jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -208,6 +208,82 @@ ecma_op_create_dynamic_function_arguments_helper (const ecma_value_t *arguments_
   return ecma_stringbuilder_finalize (&builder);
 } /* ecma_op_create_dynamic_function_arguments_helper */
 
+/**
+ * Function object creation operation.
+ *
+ * See also: ECMA-262 v5, 13.2
+ *
+ * @return pointer to newly created Function object
+ */
+static ecma_object_t *
+ecma_op_create_function_object (ecma_object_t *scope_p, /**< function's scope */
+                                const ecma_compiled_code_t *bytecode_data_p, /**< byte-code array */
+                                ecma_builtin_id_t proto_id) /**< builtin id of the prototype object */
+{
+  JERRY_ASSERT (ecma_is_lexical_environment (scope_p));
+
+  /* 1., 4., 13. */
+  ecma_object_t *prototype_obj_p = ecma_builtin_get (proto_id);
+
+  size_t function_object_size = sizeof (ecma_extended_object_t);
+
+#if ENABLED (JERRY_SNAPSHOT_EXEC)
+  if (bytecode_data_p->status_flags & CBC_CODE_FLAGS_STATIC_FUNCTION)
+  {
+    function_object_size = sizeof (ecma_static_function_t);
+  }
+#endif /* ENABLED (JERRY_SNAPSHOT_EXEC) */
+
+  ecma_object_t *func_p = ecma_create_object (prototype_obj_p,
+                                              function_object_size,
+                                              ECMA_OBJECT_TYPE_FUNCTION);
+
+  /* 2., 6., 7., 8. */
+  /*
+   * We don't setup [[Get]], [[Call]], [[Construct]], [[HasInstance]] for each function object.
+   * Instead we set the object's type to ECMA_OBJECT_TYPE_FUNCTION
+   * that defines which version of the routine should be used on demand.
+   */
+
+  /* 3. */
+  /*
+   * [[Class]] property is not stored explicitly for objects of ECMA_OBJECT_TYPE_FUNCTION type.
+   *
+   * See also: ecma_object_get_class_name
+   */
+
+  ecma_extended_object_t *ext_func_p = (ecma_extended_object_t *) func_p;
+
+  /* 9. */
+  ECMA_SET_NON_NULL_POINTER_TAG (ext_func_p->u.function.scope_cp, scope_p, 0);
+
+  /* 10., 11., 12. */
+
+#if ENABLED (JERRY_SNAPSHOT_EXEC)
```

Comment:
In previous PR we used a bit different style to avoid guarding the last `}`:
```c
#if ENABLED(...)
if (...)
{
 ...
}
else
#endif
{
}
```

Reply:
Ahh I see this function was moved. Still, maybe it's a good time to change the guards .